### PR TITLE
Simplify connection design & add Find Contents/Find Similar to Exa MCP

### PR DESCRIPTION
**Pull request by arjunkmrm**
**Refactored agent configuration to provide a cleaner and more intuitive API for setting up MCP servers**
- Updated the Connection.connect() method to accept direct server instances, reducing nesting and improving readability.
- Modified type definitions to support both direct and wrapped server configurations, ensuring flexibility and type safety.
- Moved the timeout parameter to the callTools method options, aligning with best practices for request-specific configurations.

### CODE REVIEW

File: mcps/agent/src/index.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const ConfigSchema = z.object({
 	model: z
 		.enum(["claude-3-5-sonnet-20241022"])
 		.optional()
-		.describe("The model to use. Default to 'claude-3-5-sonnet-20241022'."),
+		.describe("The model to use. Default to 'claude-3-5-sonnet-20241022."),
```
Comment (calclavia) [inline mcps/agent/src/index.ts:69]: This remove seems wrong

File: examples/test-agent.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
+// NOTE: AGENT HAS ACCESS TO FS
```
Comment (calclavia) [inline examples/test-agent.ts:1]: Uncommit this

Comment (arjunkmrm) [reply examples/test-agent.ts]: done

File: mcps/agent/src/index.ts (hunk 1 above)
Comment (arjunkmrm) [reply mcps/agent/src/index.ts]: fixed this

**Pull request merged**